sentry-ruby/lib/sentry/transaction.rb

```diff
@@ -123,15 +126,16 @@ def set_initial_sample_decision(sampling_context:, configuration: Sentry.configu
     end
 
     def finish(hub: nil)
-      super() # Span#finish doesn't take arguments
+      hub ||= @hub || Sentry.get_current_hub
```

Comment:
Now we probably don't need `finish` to take a `hub` as argument.

Ah, okay, I see that [in Python the `finish` method takes a `hub`](https://github.com/getsentry/sentry-python/blob/19fa43fec5a20b3561a16970ce395c93ac1be57d/sentry_sdk/tracing.py#L408-L417). In JavaScript it doesn't, and the [actual hub associated with the transaction is defined at initialization time](https://github.com/getsentry/sentry-javascript/blob/438ed91bd457f897da89c7728fd074bbef0a0ce1/packages/tracing/src/transaction.ts#L22-L38).

The important things are:

1. Transactions know about the hub, Spans don't need to/should not know about the hub.
2. The hub association typically should not change during the lifetime of the transaction, so:
  a. we shouldn't need to check `@hub || Sentry.get_current_hub` in multiple places.
  b. methods on span/transaction (not the constructor) should not take a hub as argument.

Reply:
in `sentry-python`, even Span holds a hub for adding breadcrumbs. I don't think `sentry-ruby` needs that right now. but I don't want to remove the hub from span and needs to add it back for a feature laster.

https://github.com/getsentry/sentry-python/blob/19fa43fec5a20b3561a16970ce395c93ac1be57d/sentry_sdk/tracing.py#L409-L426

also, I generally use `sentry-python` as a reference, cause I think it should be the most advanced one. but from what I see from your recent comments, `sentry-javascript` seems to be a better sample?

Reply:
They've unfortunately diverged in certain ways, and while Python was first in some parts of the API, JavaScript was first in other parts.

Reply:
I see. in that case, I'll keep aligning with `sentry-python` as the language characteristics and applications are more similar.